mesonbuild/interpreter.py

```diff
@@ -3657,8 +3657,17 @@ def func_configure_file(self, node, args, kwargs):
         # Install file if requested, we check for the empty string
         # for backwards compatibility. That was the behaviour before
         # 0.45.0 so preserve it.
-        idir = kwargs.get('install_dir', None)
-        if isinstance(idir, str) and idir:
+        idir = kwargs.get('install_dir', '').strip()
```

Comment:
We must *never* touch the user's `install_dir` value like this. This will change `/usr/foo/bar ` to `/usr/foo/bar` which is wrong.